scripts/wallet.js

```diff
@@ -365,13 +365,180 @@ export class Wallet {
     }
+
+    /**
+     * Get the debit of a transaction in satoshi
+     * @param {Transaction} tx
+     */
+    getDebit(tx) {
+        let debit = 0;
+        for (const vin of tx.vin) {
+            if (mempool.txmap.has(vin.outpoint.txid)) {
+                const spentVout = mempool.txmap.get(vin.outpoint.txid).vout[
+                    vin.outpoint.n
+                ];
+                if (
+                    (this.isMyVout(spentVout.script) &
+                        UTXO_WALLET_STATE.SPENDABLE_TOTAL) !=
+                    0
+                ) {
+                    debit += spentVout.value;
+                }
+            }
+        }
+        return debit;
+    }
+
+    /**
+     * Get the credit of a transaction in satoshi
+     * @param {Transaction} tx
+     */
+    getCredit(tx, filter) {
+        let credit = 0;
+        for (const vout of tx.vout) {
+            if ((this.isMyVout(vout.script) & filter) != 0) {
+                credit += vout.value;
+            }
+        }
+        return credit;
+    }
+
+    /**
+     * Return true if the transaction contains undelegations regarding the given wallet
+     * @param {Transaction} tx
+     */
+    checkForUndelegations(tx) {
+        for (const vin of tx.vin) {
+            if (mempool.txmap.has(vin.outpoint.txid)) {
+                const spentVout = mempool.txmap.get(vin.outpoint.txid).vout[
+                    vin.outpoint.n
+                ];
+                if (
+                    (this.isMyVout(spentVout.script) &
+                        UTXO_WALLET_STATE.SPENDABLE_COLD) !=
+                    0
+                ) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Return true if the transaction contains delegations regarding the given wallet
+     * @param {Transaction} tx
+     */
+    checkForDelegations(tx) {
+        for (const vout of tx.vout) {
+            if (
+                (this.isMyVout(vout.script) &
+                    UTXO_WALLET_STATE.SPENDABLE_COLD) !=
+                0
+            ) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Return the output addresses for a given transaction
+     * @param {Transaction} tx
+     */
+    getOutAddress(tx) {
+        let addresses = [];
+        for (const vout of tx.vout) {
+            const dataBytes = hexToBytes(vout.script);
+            if (isP2PKH(dataBytes)) {
+                addresses.push(
+                    this.getAddressFromHashCache(
+                        bytesToHex(
+                            dataBytes.slice(
+                                P2PK_START_INDEX,
+                                P2PK_START_INDEX + 20
+                            )
+                        ),
+                        false
+                    )
+                );
+            } else if (isP2CS(dataBytes)) {
+                for (let i = 0; i < 2; i++) {
```

Comment:
This does almost the same thing as `isMyVout`, seems like `isMyVout` can use the implementation of this one to avoid the copy-paste

Reply:
at the moment I don't see a way to merge the two functions